src/main/rx/cc2500_common.c

```diff
@@ -0,0 +1,229 @@
+ *
+ * Cleanflight and Betaflight are distributed in the hope that they
+ * will be useful, but WITHOUT ANY WARRANTY; without even the implied
+ * warranty of MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.
+ * See the GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this software.
+ *
+ * If not, see <http://www.gnu.org/licenses/>.
+ */
+
+#include <stdbool.h>
+
+#include "platform.h"
+
+#if defined(USE_RX_FRSKY_SPI) || defined(USE_RX_SFHSS_SPI)
+
+#include "common/maths.h"
+
+#include "pg/pg.h"
+#include "pg/pg_ids.h"
+#include "pg/rx.h"
+#include "pg/rx_spi.h"
+
+#include "drivers/rx/rx_cc2500.h"
+#include "drivers/io.h"
+#include "drivers/time.h"
+
+#include "fc/config.h"
+
+#include "rx/rx.h"
+#include "rx/rx_spi.h"
+
+#include "rx/cc2500_common.h"
+
+static IO_t gdoPin;
+static IO_t bindPin = DEFIO_IO(NONE);
+static IO_t cc2500LedPin;
+static bool bindRequested;
+
+static bool lastBindPinStatus;
+#if defined(USE_RX_CC2500_SPI_PA_LNA)
+static IO_t txEnPin;
+static IO_t rxLnaEnPin;
+static IO_t antSelPin;
+#endif
+static int16_t rssiDbm;
+
+uint16_t cc2500getRssiDbm(void)
+{
+    return rssiDbm;
+}
+
+void cc2500setRssiDbm(uint8_t value)
+{
+    if (value >= 128) {
+        rssiDbm = ((((uint16_t)value) * 18) >> 5) - 82;
+    } else {
+        rssiDbm = ((((uint16_t)value) * 18) >> 5) + 65;
+    }
+
+    setRssi(rssiDbm << 3, RSSI_SOURCE_RX_PROTOCOL);
+}
+
+void cc2500SpiBind(void)
+{
+    bindRequested = true;
+}
+
+bool cc2500checkBindRequested(bool reset)
+{
+    if (bindPin) {
+        bool bindPinStatus = IORead(bindPin);
+        if (lastBindPinStatus && !bindPinStatus) {
+            bindRequested = true;
+        }
+        lastBindPinStatus = bindPinStatus;
+    }
+
+    if (!bindRequested) {
+        return false;
+    } else {
+        if (reset) {
+            bindRequested = false;
+        }
+
+        return true;
+    }
+}
+
+bool cc2500getGdo(void)
+{
+    return IORead(gdoPin);
+}
+
+#if defined(USE_RX_CC2500_SPI_PA_LNA) && defined(USE_RX_CC2500_SPI_DIVERSITY)
+void cc2500switchAntennae(void)
+{
+    static bool alternativeAntennaSelected = true;
+
+    if (alternativeAntennaSelected) {
+        IOLo(antSelPin);
+    } else {
+        IOHi(antSelPin);
+    }
+    alternativeAntennaSelected = !alternativeAntennaSelected;
+}
+#endif
+#if defined(USE_RX_CC2500_SPI_PA_LNA)
+void cc2500TxEnable(void)
+{
+    IOHi(txEnPin);
+}
+
+void cc2500TxDisable(void)
+{
+    IOLo(txEnPin);
+}
+#endif
+
+void cc2500LedOn(void)
+{
+#if defined(RX_CC2500_SPI_LED_PIN_INVERTED)
+    IOLo(cc2500LedPin);
+#else
+    IOHi(cc2500LedPin);
+#endif
+}
+
+void cc2500LedOff(void)
+{
+#if defined(RX_CC2500_SPI_LED_PIN_INVERTED)
+    IOHi(cc2500LedPin);
+#else
+    IOLo(cc2500LedPin);
+#endif
+}
+
+void cc2500LedBlink(timeMs_t blinkms)
+{
+ static bool ledIsOn=true;
+ static timeMs_t ledBlinkMs = 0;
+
+    if ( (ledBlinkMs + blinkms) > millis() ){
```

Comment:
No spaces directly on the inside of `(`.

Reply:
fixed